File: Go/isMonotonic.go
package main

func isMonotonic(nums []int) bool {

	if len(nums) == 0 {
		return false
	}

	increase := true
	decrese := true

	for i := 0; i < len(nums)-1; i++ {
		if nums[i] > nums[i+1] {
			increase = false
		}
		if nums[i] < nums[i+1] {
			decrese = false
		}
	}

	return increase || decrese
}

File: data structures/C#/IsMonotonic.cs
//O(n)
//O(1)
public class Solution
{
    public bool IsMonotonic(int[] nums)
    {
        if (nums.Length == 0) return true;

        bool increasing = true;
        bool decreasing = true;

        for (int i = 0; i < nums.Length - 1; i++)
        {
            if (nums[i] < nums[i + 1])
                increasing = false;
            if (nums[i] > nums[i + 1])
                decreasing = false;
        }

        return increasing || decreasing;
    }
}